modules/adformBidAdapter.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'adform';
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [],
+  isBidRequestValid: function (bid) {
+    return !!(bid.params.mid);
+  },
+  buildRequests: function (validBidRequests) {
+    var i, l, j, k, bid, _key, _value, reqParams;
+    var request = [];
+    var globalParams = [ [ 'adxDomain', 'adx.adform.net' ], [ 'fd', 1 ], [ 'url', null ], [ 'tid', null ] ];
+    var netRevenue = 'net';
+    for (i = 0, l = validBidRequests.length; i < l; i++) {
+      bid = validBidRequests[i];
+      if (bid.params.priceType === 'gross') {
+        netRevenue = 'gross';
+      }
+      for (j = 0, k = globalParams.length; j < k; j++) {
+        _key = globalParams[j][0];
+        _value = bid[_key] || bid.params[_key];
+        if (_value) {
+          bid[_key] = bid.params[_key] = null;
+          globalParams[j][1] = _value;
+        }
+      }
+      reqParams = bid.params;
+      reqParams.transactionId = bid.transactionId;
+      request.push(formRequestUrl(reqParams));
+    }
+
+    request.unshift('//' + globalParams[0][1] + '/adx/?rp=4');
+
+    request.push('auctionId=' + validBidRequests[0].requestId);
+
+    for (i = 1, l = globalParams.length; i < l; i++) {
+      _key = globalParams[i][0];
+      _value = globalParams[i][1];
+      if (_value) {
+        request.push(_key + '=' + encodeURIComponent(_value));
+      }
+    }
+
+    return {
+      method: 'GET',
+      url: request.join('&'),
+      bids: validBidRequests,
+      netRevenue: netRevenue,
+      bidder: 'adform'
+    };
+
+    function formRequestUrl(reqData) {
+      var key;
+      var url = [];
+
+      for (key in reqData) {
+        if (reqData.hasOwnProperty(key) && reqData[key]) { url.push(key, '=', reqData[key], '&'); }
+      }
+      return encode64(url.join('').slice(0, -1));
+    }
+
+    function encode64(input) {
```

Comment:
is there a reason you are manually performing base64 encoding over using native `btoa`?